config/include.go

```diff
@@ -35,7 +35,16 @@ func parseIncludedConfig(
 		includePath = util.JoinPath(filepath.Dir(terragruntOptions.TerragruntConfigPath), includePath)
 	}
 
-	if len(decodeList) > 0 {
+	// We need to do a partial parse if we are in a parsing stage that is performing a partial parse to honor the
+	// configuration parsing order. However, strictly speaking, the only case where we can't do a full parse of the
+	// included config is if the included config has any `dependency` or `dependencies` blocks defined. So any config
+	// that does not have any dependency or dependencies blocks can be fully parsed without issue, regardless of parsing
+	// stage.
+	hasDependency, err := configFileHasDependencyBlock(includePath, terragruntOptions)
+	if err != nil {
+		return nil, err
+	}
+	if hasDependency && len(decodeList) > 0 {
```

Comment:
> I thought the comment above says we can only do a partial parse if there isn't a dependency or dependencies block... So should this bool check be inverted to !hasDependency?

Ah I meant actually the exact opposite in the comment (the code is correct). I reworded the comment, and hopefully the new version makes that more clear: 8d5638c

Reply:
> Should we show any kind of warning or log message if hasDependency is true, so users know what's going on?

I opted not to here, mainly because it's really hard to understand this nuance from just the logs. I think the best way to make this obvious to the user is to implement some kind of terragrunt language server and have it show up as an editor warning.

Reply:
But will this ultimately lead to a confusing error for the user? Where they are referencing some output that isn't defined yet?

Reply:
Ok I added a warning log when the partial parsing happens: bd4288a8a4e5ad8dc9fd1f21283abf7d067574c4